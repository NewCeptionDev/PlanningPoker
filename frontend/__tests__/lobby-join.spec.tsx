import { act, render, screen, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import LobbyJoin from 'src/app/[slug]/page'

jest.mock('next/navigation', () => ({
  useRouter: () => {
    return {
      prefetch: () => {},
    }
  },
}))

describe('LobbyJoin', () => {
  beforeAll(() => {
    // @ts-expect-error only partially defined
    const response: Response = {
      ok: true,
      json: async () => {
        return {
          lobbyExists: true,
          lobbyName: 'Test',
        }
      },
    }
    global.fetch = jest.fn().mockResolvedValue(response)
  })

  beforeEach(() => {
    jest.resetModules()
  })

  it('should render correctly', () => {
    const { container } = render(<LobbyJoin params={{ slug: '1234' }} />)
    expect(container).toMatchSnapshot()
  })

  it('should render all important components', () => {
    render(<LobbyJoin params={{ slug: '1234' }} />)
    expect(screen.getByTestId('logo')).toBeInTheDocument()
    expect(screen.getByTestId('headline')).toBeInTheDocument()
    expect(screen.getByTestId('lobby-name')).toBeInTheDocument()
    expect(screen.getByTestId('theme-switcher')).toBeInTheDocument()
    expect(screen.getByTestId('lobby-join')).toBeInTheDocument()
    expect(screen.getByTestId('footer')).toBeInTheDocument()
  })

  it('should render correct lobby name', async () => {
    render(<LobbyJoin params={{ slug: '1234' }} />)

    await waitFor(() => expect(screen.getByTestId('lobby-name')).toHaveTextContent('Test'))
  })
})
