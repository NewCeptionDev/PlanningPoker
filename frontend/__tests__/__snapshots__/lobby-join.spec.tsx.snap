// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LobbyJoin should render correctly 1`] = `
<div>
  <div
    class="flex flex-row h-[5vh] justify-between p-4 secondary"
  >
    <div
      class="w-1/5 flex items-center"
    >
      <div
        class="flex flex-row cursor-pointer"
      >
        <img
          alt="logo"
          class="mr-4"
          data-nimg="1"
          data-testid="logo"
          decoding="async"
          height="21"
          loading="lazy"
          src="/_next/image?url=%2Flogo.png&w=32&q=75"
          srcset="/_next/image?url=%2Flogo.png&w=16&q=75 1x, /_next/image?url=%2Flogo.png&w=32&q=75 2x"
          style="color: transparent;"
          width="16"
        />
        <h1
          class="text-white"
          data-testid="headline"
        >
          Planning Poker
        </h1>
      </div>
    </div>
    <div
      class="w-3/5 flex items-center justify-center"
    >
      <h1
        class="text-white"
        data-testid="lobby-name"
      />
    </div>
    <div
      class="flex flex-row justify-end w-1/5 items-center"
    >
      <div
        class="grid place-items-center h-full relative"
        data-testid="theme-switcher"
      >
        <input
          class="hidden toggle--checkbox"
          data-testid="theme-switcher-toggle"
          id="toggle"
          type="checkbox"
        />
        <label
          class="toggle--label"
          data-testid="theme-switcher-label"
          for="toggle"
        >
          <span
            class="toggle--label-background"
            data-testid="theme-switcher-label-background"
          />
        </label>
      </div>
      <button
        class="btn ml-8"
      >
        Leave
      </button>
    </div>
  </div>
  <div
    class="flex flex-col items-center h-[91vh] justify-center"
    data-testid="lobby-join"
  >
    <h1>
      Enter your name
    </h1>
    <form
      class="flex flex-col items-center m-4"
    >
      <input
        class="input"
        name="name"
        placeholder="Name"
        type="text"
      />
      <p
        class="m-4"
      >
        Select your role
      </p>
      <select
        class="input"
        name="role"
      >
        <option
          class="text-black dark:text-white"
          value="PLAYER"
        >
          Player
        </option>
        <option
          class="text-black dark:text-white"
          value="OBSERVER"
        >
          Observer
        </option>
      </select>
      <button
        class="btn mt-4"
        type="submit"
      >
        Join
      </button>
    </form>
  </div>
  <div
    class="flex flex-row justify-center h-[2vh]"
    data-testid="footer"
  >
    <p>
      Made by 
      <a
        href="https://newception.dev"
      >
        @NewCeption
      </a>
    </p>
  </div>
</div>
`;
